### PR TITLE
Added support for naming of Indexed types and avoid passing unnecessary kw_args

### LINKED ISSUE #14960: `integrate` raised issue with variable of integration of type `Indexed`
This error was raised with the particular expression given below. 
```
from sympy import *
>>> x = Symbol('x')
>>> x1, x2 = Indexed('x', 1), Indexed('x', 2)
>>> pdf = exp(-x1**2/2 + x1 - x2**2/2 - S(1)/2)/(2*pi)
>>> integrate(pdf, (x1, -oo, oo))
...
AttributeError: 'Indexed' object has no attribute 'name'
```

### PR DESCRIPTION
**Pull request by ArifAhmed1995**
#### References to other Issues or PRs
Fixes #14960 

#### Brief description of what is fixed or changed
* IndexedBase and Indexed types have been given a name attribute. 
* kw_args are not unnecessarily passed to Expr when creating an IndexedBase object
#### Release Notes

* tensor
  * Added naming for IndexedBase and Indexed types
  * Avoid passing kw_args to Expr during IndexedBase object creation

### CODE REVIEW

**Comment by ArifAhmed1995** [comment]
@jksuom Will this suffice for fixing #14960 ?

**Comment by jksuom** [comment]
Let us see what @akash9712 has to say.

**Comment by ArifAhmed1995** [comment]
Yes.
@akash9712 I think you were facing some issue with a change in `stats` module. Does this PR address it ?

**Comment by akash9712** [comment]
Sorry for the delay. This seems to work fine with the expression. Should a test be added in for `integrate` with indexed symbol as well?

File: sympy/tensor/indexed.py [hunk 1]
```diff
@@ -159,6 +159,14 @@ def __new__(cls, base, *args, **kw_args):
 
         return Expr.__new__(cls, base, *args, **kw_args)
 
+    @property
+    def name(self):
+        indices_str = "["
+        for index in self.args[1:-1]:
+            indices_str += str(index) + ","
+        indices_str += str(self.args[-1]) + "]"
+        return self.base.name + indices_str
```
Comment (jksuom) [inline sympy/tensor/indexed.py]: Would it suffice to return `str(self)`?

Comment (ArifAhmed1995) [reply sympy/tensor/indexed.py]: Yes, it does. Missed that earlier. Updated now.

File: sympy/tensor/indexed.py [hunk 2]
```diff
@@ -403,8 +407,13 @@ def __new__(cls, label, shape=None, **kw_args):
         obj._shape = shape
         obj._offset = offset
         obj._strides = strides
+        obj.__name__ = str(label)
```
[the remark this answers is not in the document]
Comment (ArifAhmed1995) [reply sympy/tensor/indexed.py]: Changed.

**Comment by jksuom** [comment]
There is another issue that could be fixed in this PR. `IndexedBase` admits keyword arguments that are passed on to `Expr` which does not accept any.
```
>>> A = IndexedBase('A', real=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/tensor/indexed.py", line 402, in __new__
    obj = Expr.__new__(cls, label, **kw_args)
TypeError: __new__() got an unexpected keyword argument 'real'
```
Those `**kw_args` should be removed from the argument list of `Expr.__new__`.

**Comment by ArifAhmed1995** [comment]
@jksuom Okay, done.
As Akash said earlier, should the integral test be added in this PR ?
I personally think it would make more sense to add that test elsewhere since this PR isn't targeted towards solving integrals involving `Indexed` objects.

**Comment by akash9712** [comment]
If it looks fine, can this be merged now?

**Comment by jksuom** [comment]
Thanks, this looks good to me.

**Pull request merged**